bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/SonnenHandler.java

```diff
@@ -176,6 +196,29 @@ private void updateChannel(String channelId) {
         if (isLinked(channelId)) {
             State state = null;
             SonnenJsonDataDTO data = serviceCommunication.getBatteryData();
+            // String direction = GetPowerMeterDirection(channelId);
+
+            // The sonnen API has two sub-channels, e.g. 4_1 and 4_2, one representing consumption and the
+            // other production. E.g. 4_1.kwh_imported represents the total production since the
+            // battery was installed.
+            SonnenJsonPowerMeterDataDTO[] dataPM = null;
+            if (config.powerMeter) {
+                dataPM = serviceCommunication.getPowerMeterDatas();
+            }
+
+            if (dataPM != null) {
+                switch (channelId) {
+                    case ENERGYIMPORTEDSTATE:
+                        state = new QuantityType<Energy>(dataPM[1].getKwhImported(), Units.WATT_HOUR);
```

Comment:
This is really confusing: Can you confirm that the API provides values in **wh** (watt hours) when the API name is actually **kwh_** imported (kilowatt hours)?

Reply:
Yes I can. For sure. I know it is odd, however it is really the case. I changed it once and notice then that the values are incorrect.

Reply:
OK, thanks. Consider adding a short comment explaining this to avoid accidentally having this changed back to `Units.KILOWATT_HOUR` by someone thinking this is an obvious bug to fix.

Reply:
The values I am getting in "kwh_imported" from /api/v2/powermeter are in kWh: Changing from 8199.3 to 8200.3 within 15 mins while production is at approx. 4 kW. As expected for values expressed in kWh. 
Numbers seem to be at 0.1 kWh resolution. Maybe this can also be considered in the interface.

Reply:
@chingon007 - can you check this again? Perhaps you can both post the payload you are receiving?

Reply:
Ok guys, you are right. I made a mistake in the code declaring the unit at the state with KWh even it is Watt hours. I considered also 0.1 kWh resolution.

Reply:
@timbms for the values of "kwh_imported" and "kwh_exported" from /api/v2/powermeter they seem to be every day equal to me. I haven't seen a change in the value. Do I need to reset them somehow?

Reply:
Ok I fixed it. I added also two more channels for the powermeter.

Reply:
ok I fixed it. Added also two new channels for PowerMeter